app/controllers/catalog_controller.rb

```diff
@@ -370,22 +370,15 @@ def show
     authorize! :read, @cocina
 
     @workflows = WorkflowService.workflows_for(druid: @druid)
-    milestones = MilestoneService.milestones_for(druid: @druid)
-    object_client = Dor::Services::Client.object(@druid)
-    versions = object_client.version.inventory
 
     user_versions = object_client.user_version.inventory
     raise ActionController::RoutingError, 'Not Found' if @user_version && user_versions.none? { |version| version.userVersion.to_s == @user_version }
```

Comment:
Looking good. I was wondering about also doing this (plus the suggestion below):

```suggestion
    @milestones_presenter = MilestonesPresenter.new(user_version: @user_version, druid: @druid)
    raise ActionController::RoutingError, 'Not Found' if @milestones_presenter.valid_user_version?
```

Reply:
Sneaky.